### PR TITLE
Sergei / chore: migrate open-positions to tsx

### PR DESCRIPTION
**Pull request by sergei-deriv**
## Changes:

Please include a summary of the change and which issue is fixed below:
-   convert packages/reports/src/Containers/open-positions.jsx to tsx

## When you need to add unit test

-   [ ] If this change disrupt current flow
-   [ ] If this change is adding new flow

## When you need to add integration test

-   [ ] If components from external libraries are being used to define the flow, e.g. @deriv/components
-   [ ] If it relies on a very specific set of props with no default behavior for the current component.

## Test coverage checklist (for reviewer)

-   [ ] Ensure utility / function has a test case 
-   [ ] Ensure all the tests are passing

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: packages/stores/types.ts [hunk 1]
```diff
@@ -1,5 +1,7 @@
+import { TPortfolioPos } from '@deriv/shared';
 import type { GetAccountStatus, Authorize, DetailsOfEachMT5Loginid } from '@deriv/api-types';
 import type { RouteComponentProps } from 'react-router';
+import { TContractInfo } from '../shared/src/utils/contract/contract-types';
```
Comment (farzin-deriv) [inline packages/stores/types.ts]: @sergei-deriv This package shouldn’t have dependencies on other packages just to reuse some types, You should move `TPortfolioPos` and `TContractInfo` to this package at least for now.
By the way can’t we use `PortfolioResponse` from `@deriv/api-types`? 🤔

File: packages/components/src/components/contract-card/contract-card-items/contract-card-sell.jsx [hunk 2]
```diff
@@ -14,24 +14,24 @@ const ContractCardSell = ({ contract_info, getCardLabels, is_sell_requested, onC
...
-                        text={getCardLabels().SELL}
-                        onClick={onClick}
-                        secondary
-                    />
-                ) : (
-                    <div className='dc-contract-card__no-resale-msg'>{getCardLabels().RESALE_NOT_OFFERED}</div>
-                )}
-            </React.Fragment>
-        )
+    return should_show_sell ? (
```
Comment (farzin-deriv) [inline packages/components/src/components/contract-card/contract-card-items/contract-card-sell.jsx]: @sergei-deriv Not related to your PR, But since you have changed the condition I suggest you change it to the `Return Early` pattern.

File: packages/components/src/components/progress-bar/progress-bar.tsx [hunk 3]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 import Text from '../text';
 
 // Todo: Create a utility type for getting a range of float number. ex: FloatRange<0, 1>
-type TRangeFloatZeroToOne = 0 | 0.1 | 0.2 | 0.3 | 0.4 | 0.5 | 0.6 | 0.7 | 0.8 | 0.9 | 1;
+export type TRangeFloatZeroToOne = 0 | 0.1 | 0.2 | 0.3 | 0.4 | 0.5 | 0.6 | 0.7 | 0.8 | 0.9 | 1;
```
Comment (farzin-deriv) [inline packages/components/src/components/progress-bar/progress-bar.tsx]: @sergei-deriv No need to export, Please use `React.ComponentProps` generic.

Comment (sergei-deriv) [reply packages/components/src/components/progress-bar/progress-bar.tsx]: > @sergei-deriv No need to export, Please use `React.ComponentProps` generic.

I've added this type in 'open-positions' component:
type TRangeFloatZeroToOne = React.ComponentProps<typeof ProgressBar>['value'];

File: packages/components/src/components/contract-card/contract-card-items/contract-card-sell.jsx (hunk 2 above)
Comment (sergei-deriv) [reply packages/components/src/components/contract-card/contract-card-items/contract-card-sell.jsx]: > @sergei-deriv Not related to your PR, But since you have changed the condition I suggest you change it to the `Return Early` pattern.

I did it because with syntax **"return should_show_sell && ( JSX )"** this component returned JSX of False. But with my refactored code **"return should_show_sell ? ( JSX ) : (JSX)"** this component always returns JSX.

File: packages/stores/types.ts (hunk 1 above)
Comment (sergei-deriv) [reply packages/stores/types.ts]: > @sergei-deriv This package shouldn’t have dependencies on other packages just to reuse some types, You should move `TPortfolioPos` and `TContractInfo` to this package at least for now. By the way can’t we use `PortfolioResponse` from `@deriv/api-types`? 🤔

I changed TPortfolioPos to PortfolioResponse.

File: packages/components/src/components/progress-bar/progress-bar.tsx [hunk 4]
```diff
@@ -3,7 +3,7 @@ import React from 'react';
 import Text from '../text';
 
 // Todo: Create a utility type for getting a range of float number. ex: FloatRange<0, 1>
```
Comment (maryia-deriv) [inline packages/components/src/components/progress-bar/progress-bar.tsx]: @sergei-deriv does this PR also implies that we address the TODO comment? 🙂 if yes, we can make a TFloatRange type according to an example in https://catchts.com/range-numbers

Comment (sergei-deriv) [reply packages/components/src/components/progress-bar/progress-bar.tsx]: > @sergei-deriv does this PR also implies that we address the TODO comment? 🙂 if yes, we can make a TFloatRange type according to an example in https://catchts.com/range-numbers

I'm not sure, honestly :) But if you think I have to do this, I'll try. But for my PR I've got this type like this:
type TRangeFloatZeroToOne = React.ComponentProps<typeof ProgressBar>['value'];

File: packages/components/src/components/contract-card/contract-card-items/contract-card-sell.jsx (hunk 2 above)
Comment (farzin-deriv) [reply packages/components/src/components/contract-card/contract-card-items/contract-card-sell.jsx]: @sergei-deriv You can use the `Return Early` pattern for better readability, It's still the same.

```tsx
if (!should_show_sell) return <></>;
if (!is_valid_to_sell) return <div...
return <Button...
```

It's just a suggestion tho 😅 Feel free to ignore it 🙇🏻

File: packages/components/src/components/progress-bar/progress-bar.tsx (hunk 4 above)
Comment (farzin-deriv) [reply packages/components/src/components/progress-bar/progress-bar.tsx]: I think it’s out of this PR scope to handle the todo, But it will be nice if we add the generic @maryia-deriv suggested to `utils.d.ts` 👍🏻 Maybe in a separate card 🤔

File: packages/components/src/components/contract-card/contract-card-items/contract-card-sell.jsx (hunk 2 above)
Comment (sergei-deriv) [reply packages/components/src/components/contract-card/contract-card-items/contract-card-sell.jsx]: > @sergei-deriv You can use the `Return Early` pattern for better readability, It's still the same.
> 
> ```tsx
> if (!should_show_sell) return <></>;
> if (!is_valid_to_sell) return <div...
> return <Button...
> ```
> 
> It's just a suggestion tho 😅 Feel free to ignore it 🙇🏻

It's really good advice, I fully agree with you. I've updated my commit.

File: packages/shared/src/utils/helpers/format-response.ts [hunk 5]
```diff
@@ -1,24 +1,74 @@
...
-    transaction_ids?: {
-        buy: number;
-        sell: number;
-    };
-    limit_order?: {
-        stop_loss?: null | number;
-        take_profit?: null | number;
-    };
-};
+// export type TPortfolioPos = {
```
Comment (mahdiyeh-deriv) [inline packages/shared/src/utils/helpers/format-response.ts]: please remove commented lines

File: packages/shared/src/utils/helpers/format-response.ts [hunk 6]
```diff
@@ -30,24 +80,29 @@ const isUnSupportedContract = (portfolio_pos: TIsUnSupportedContract) =>
...
     const purchase = portfolio_pos.buy_price;
     const payout = portfolio_pos.payout;
-    const display_name = getSymbolDisplayName(active_symbols, getMarketInformation(portfolio_pos.shortcode).underlying);
-    const transaction_id =
-        portfolio_pos.transaction_id || (portfolio_pos.transaction_ids && portfolio_pos.transaction_ids.buy);
+    const display_name = getSymbolDisplayName(
+        active_symbols,
+        getMarketInformation(portfolio_pos.shortcode ?? '').underlying
+    );
+    // const transaction_id =
```
Comment (mahdiyeh-deriv) [inline packages/shared/src/utils/helpers/format-response.ts]: why did you change this:
`        portfolio_pos.transaction_id || (portfolio_pos.transaction_ids && portfolio_pos.transaction_ids.buy);`

File: packages/shared/src/utils/helpers/format-response.ts (hunk 5 above)
Comment (sergei-deriv) [reply packages/shared/src/utils/helpers/format-response.ts]: Done

File: packages/shared/src/utils/helpers/format-response.ts (hunk 6 above)
Comment (sergei-deriv) [reply packages/shared/src/utils/helpers/format-response.ts]: > why did you change this: ` portfolio_pos.transaction_id || (portfolio_pos.transaction_ids && portfolio_pos.transaction_ids.buy);`

It's because type 'ProposalOpenContract' doesn't have property 'transaction_id', only 'transaction_ids'.

File: packages/stores/types.ts [hunk 7]
```diff
@@ -1,4 +1,5 @@
-import type { GetAccountStatus, Authorize, DetailsOfEachMT5Loginid } from '@deriv/api-types';
+import moment from 'moment';
```
Comment (farzin-deriv) [inline packages/stores/types.ts]:
```suggestion
import type { Moment } from 'moment';
```
@sergei-deriv Also you need to install `moment` as a dev dependency in `@deriv/stores`

Comment (sergei-deriv) [reply packages/stores/types.ts]: > @sergei-deriv Also you need to install `moment` as a dev dependency in `@deriv/stores`

Done. Thanks for your help

**Review by farzin-deriv — APPROVED** [review]

File: packages/stores/types.ts (hunk 7 above)
Comment (mahdiyeh-deriv) [reply packages/stores/types.ts]: @sergei-deriv please check the circlie-ci issue

File: packages/components/src/components/contract-card/contract-card-items/contract-card-sell.jsx [hunk 8]
```diff
@@ -14,24 +14,19 @@ const ContractCardSell = ({ contract_info, getCardLabels, is_sell_requested, onC
         ev.preventDefault();
     };
 
+    if (!should_show_sell) return <React.Fragment />;
```
Comment (mahdiyeh-deriv) [inline packages/components/src/components/contract-card/contract-card-items/contract-card-sell.jsx]: why we are returning empty fragment? isn't to return `null` instead?

Comment (sergei-deriv) [reply packages/components/src/components/contract-card/contract-card-items/contract-card-sell.jsx]: > why we are returning empty fragment? isn't to return `null` instead?

I did it because component is waiting JSX. I'll change it now

**Review by mahdiyeh-deriv — APPROVED** [review]
